Fix/update challenges for basic html and html5

aditijoshi14 opened the pull request:
Currently, for this following challenge:
https://learn.freecodecamp.org/responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list/
An empty string was being accepted. With this fix, it will not allow an empty string or whitespaces as an input. 

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (only English)
- [X] My changes do not use shortened URLs or affiliate links.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes part of issue #18007

joshiad committed:
fix: challenge for basic html and html5 - test

aditijoshi14 committed:
Delete use-if-and-else-to-add-logic-to-your-styles.english.md

RandellDawson commented:
@aditijoshi14 You are probably wondering why you have a merge conflict.  You probably need to run:
`git pull --rebase upstream master` on your local branch with the changes and then resolve the conflict if it still exists.

joshiad committed:
fix: Resolving merge conflict

aditijoshi14 commented:
@RandellDawson Thank you so much I was able to resolve the conflict.

RandellDawson commented:
I don't think you meant to have that other file on the PR did you?

aditijoshi14 commented:
Nope. I didn't mean to have the other file on this PR but I had to commit that to resolve the conflict. I apologize for the confusion.

RandellDawson commented on curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list.english.md:
```diff
@@ -32,7 +32,8 @@ tests:
     testString: assert(code.match(/<\/li>/gi) && code.match(/<li[\s>]/gi) && code.match(/<\/li>/gi).length === code.match(/<li[\s>]/gi).length, 'Make sure your <code>li</code> elements have closing tags.');
-
+  - text: Make sure your <code>li</code> elements doesn’t contain empty string or only whitespace.
+    testString: assert($("ul li").filter(function(index){return $(this).text().trim().length === 0;}).length === 0, 'Make sure your <code>li</code> elements doesn\’t contain empty string or only whitespace.');
```
While your added test works fine, yo could use the same logic but simplify a bit.
```suggestion
    testString: assert($("ul li").filter((_, item) => !$(item).text().trim()).length === 0, 'Make sure your <code>li</code> elements don\’t contain an empty string or only white-space.');
```

RandellDawson commented on curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/create-a-bulleted-unordered-list.english.md:
```diff
@@ -32,7 +32,8 @@ tests:
   - text: Make sure your <code>li</code> elements have closing tags.
     testString: assert(code.match(/<\/li>/gi) && code.match(/<li[\s>]/gi) && code.match(/<\/li>/gi).length === code.match(/<li[\s>]/gi).length, 'Make sure your <code>li</code> elements have closing tags.');
-
+  - text: Make sure your <code>li</code> elements doesn’t contain empty string or only whitespace.
```
Please make the following changes to the text.
```suggestion
  - text: Make sure your <code>li</code> elements don’t contain an empty string or only white-space.
```

RandellDawson reviewed (changes requested):
Please see my requested changes.

RandellDawson commented:
Please see my suggestions above and remove the changes to that extra file from this PR.

Thank you.

RandellDawson commented:
You still have the other file on this PR.  In the future, before you start making changes, do a rebase on your master branch `git pull --rebase upstream master` and then checkout a new branch (`git checkout -b branchname`) before making changes for your next PR.  It appears this PR has some changes left over from another branch you made for another PR.

RandellDawson commented:
That is why your commit history for this PR goes all the way back to December 6, 2018.

aditijoshi14 commented:
I apologize for that. I will surely do that from next time. Do you happen to know if there is anyway I can delete that extra file from this PR?

aditijoshi14 commented:
If you want I can create a new branch, add changes into it and make a new pull request.

RandellDawson commented:
No, just keep this PR open and I will try to help you out.  It is a little more complicated, because you have made several other commits trying to resolve this and I get merge conflicts when I try to rebase your PR locally.  In the end it will be a good learning experience for both of us.  I don't have time to debug this now, but will reply later when I have some more time to work on this.

RandellDawson commented:
What happens if you `git pull --rebase upstream master on your local `ix/update-challenges-for-basic-html-and-html5` branch?  Do you get a merge conflict?

aditijoshi14 commented:
Yes, I get a marge conflict.

joshiad committed:
fix: basic-html-and html5 challenge - test

aditijoshi14 commented:
@RandellDawson  I was able to resolve the merge conflict and remove the extra file. Thank you so much for all your help!

RandellDawson commented:
Just out of curiosity, how did you finally remove it?  I figured out two ways to do it, but you beat me to it.

RandellDawson commented:
Can you visit the Contributer's channel?  I need to discuss something with you.

aditijoshi14 commented:
@RandellDawson  I actually synced my forked repo with FreeCodeCamp's repo, which I believe was giving me merge conflict in my third commit. My third commit was when I deleted the extra file from my PR directly. After syncing my forked repo, I reverted to the third commit and made the changes you suggested. 

I am currently in Contributor's channel in Gitter.

RandellDawson commented:
I have tested the changes locally and validate the new test added works as intended without any "side-effects" that I can see.

ezioda004 commented:
Thank you @aditijoshi14 for your contribution and congratulations on your first PR merge to this repo! 🎉
Hope to see more contributions from you in the future.
